packages/discord.js/src/client/actions/MessageCreate.js

```diff
@@ -6,7 +6,11 @@ const Events = require('../../util/Events');
 class MessageCreateAction extends Action {
   handle(data) {
     const client = this.client;
-    const channel = this.getChannel({ id: data.channel_id, guild_id: data.guild_id, author: data.author });
+    const channel = this.getChannel({
+      id: data.channel_id,
+      author: data.author,
+      ...(data.guildId && { guild_id: data.guild_id }),
```

Comment:
This behavior is documented only for `MESSAGE_CREATE`. I don't think it's needed for other events

Reply:
Or it should be applied for each property

Reply:
It's documented for MESSAGE_UPDATE too, for MESSAGE_DELETE implicitly. For the others it's also documented to be optional.